File: ant-sdk/antd-lua/examples/06-private-data.lua
--- Example: Store and retrieve private (encrypted) data.

local antd = require("antd")

local client = antd.new_client()

-- Store private data (encrypted on the network). The DataMap is returned to
-- the caller; it is NOT stored on-network.
local secret = "This is my private data"
local result, err = client:data_put(secret)
if err then
    print("Put error: " .. err.message)
    os.exit(1)
end

print("Private data stored")
print("Data map: " .. result.data_map)
print(string.format("Chunks: %d, mode: %s", result.chunks_stored, result.payment_mode_used))

-- Retrieve private data using the caller-held DataMap.
local retrieved, err2 = client:data_get(result.data_map)
if err2 then
    print("Get error: " .. err2.message)
    os.exit(1)
end

print("Retrieved: " .. retrieved)

File: ant-sdk/antd-elixir/examples/06_private_data.exs
Mix.install([
  {:antd, path: ".."}
])

# Store and retrieve private (encrypted) data.
#
# Private methods are the unqualified verb: `data_put` / `data_get`. The
# returned DataMap is the caller-held handle (NOT stored on-network).
client = Antd.Client.new()

# Store private data — returns a data_map needed for retrieval.
{:ok, result} = Antd.Client.data_put(client, "my secret message", payment_mode: :auto)
IO.puts("Private data stored")
IO.puts("Data map (save this!): #{result.data_map}")
IO.puts("Chunks stored: #{result.chunks_stored}, mode: #{result.payment_mode_used}")

# Retrieve private data using the data map.
{:ok, data} = Antd.Client.data_get(client, result.data_map)
IO.puts("Retrieved: #{data}")

unless data == "my secret message", do: raise "Round-trip mismatch!"
IO.puts("Private data round-trip OK!")
